rfcs/0051-stop-transaction-processing.md

```diff
@@ -0,0 +1,146 @@
+# Stop processing transactions after a certain slot
+
+This PR describes the feature to stop processing transactions after a certain
+slot, to be used in the Berkeley hard fork.
+
+## Summary
+
+Transactions come from a client or the gossip network and are processed by BPs
+and SNARK workers to be included in blocks.
+
+In this RFC, the procedure to stop processing any new transactions after a
+certain slot is described. This is, any blocks produced after that slot will
+include no transaction at all.
+
+## Motivation
+
+In a hard fork scenario, we want to halt the preceding network and produce a new
+genesis ledger for the succeeding network. We want this to happen by defining a
+point in time (slot) where the network stops accepting transactions a keeps
+producing blocks free of transactions until it reaches a consensus state.
+
+This feature enables part of this procedure, by allowing users to define a slot
+at which the node will stop accepting any new transactions or including them in
+blocks (in case it is a BP node).
+
+## Detailed design
+
+The procedure to stop processing transactions after a certain slot will act on
+different components of the node, namely, the CLI, the transaction pool, the
+snark pool, and the block producer.
+and the CLI. The procedure will be as follows:
+
+* The CLI will allow a slot at which the node will stop processing transactions
+  to be set. This will be done by setting a new optional flag `--slot-tx-end`
+  that will be used by the transaction pool, snark pool, and the block producer
+  to stop processing transactions after a certain slot. If the flag is not set,
+  the node will continue processing transactions as usual.
+* The node will stop accepting new transactions from the client after the slot
+  set by the CLI flag.
+* The transaction pool will stop accepting new transactions after the slot set
+  by the CLI flag.
+* The transaction pool will drop all transactions from the pool after the slot
+  set by the CLI flag.
+* The snark pool will stop accepting new snark works after the slot set by the
+  CLI flag.
+* The snark pool will drop all snark works from the pool after the slot set by
```

Comment:
Isn't the point of continuing the network without transactions to get a snarked ledger that witnesses the state of the ledger? If so, we need snark work to do this, so we will need to continue operating the snark pool.

Reply:
The hard fork mechanism plan states that we will use the staged ledger to produce the new genesis ledger. In this case, we see no reason to continue snark work, since it would be paid a 0 zero fee (as mentioned in another comment) and the resulting snarked ledger would not be used to produce the new ledger.